src/models/AbstractCatalogProfile/AbstractCatalogProfile.ts

```diff
@@ -295,22 +295,28 @@ export abstract class AbstractCatalogProfileStore {
     @action setColumnFilter = (filter: string, columnName: string) => {
         const current = this.catalogControlHeader.get(columnName);
         const newHeader: ControlHeader = {
-            columnIndex: current.columnIndex,
-            dataIndex: current.dataIndex,
-            display: current.display,
+            columnIndex: current?.columnIndex ?? NaN,
+            dataIndex: current?.dataIndex ?? NaN,
+            display: current?.display ?? false,
             filter: filter,
-            columnWidth: current.columnWidth
+            columnWidth: current?.columnWidth ?? null
         };
         this.catalogControlHeader.set(columnName, newHeader);
         this.updateTableStatus(true);
     };
 
     @action.bound setTableColumnWidth(width: number, columnName: string) {
-        this.catalogControlHeader.get(columnName).columnWidth = width;
+        const header = this.catalogControlHeader.get(columnName);
+        if (header) {
```

Comment:
Can we use has() function? If(Map.has(key)) then Map.get(key) = width

Reply:
I agree it's more natural to use `Map.has` here, but unfortunately TS doesn't support type narrowing in this way. Looks like it's a known [issue](https://github.com/microsoft/TypeScript/issues/13086).
Error message:
<img width="1040" alt="image" src="https://github.com/CARTAvis/carta-frontend/assets/43841102/7ebf6659-e62b-434b-8e9f-632a0a52a0cb">